### DISCUSSION TITLE
Restart the simulation problem (loading Auxvariable - SECOND/MONOMIAL)

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#17574] [Q&A General] [answered]
Hi MOOSE users,

I've been trying to make my simulation restartable.
[](https://mooseframework.inl.gov/application_usage/restart_recover.html)

I have set my 'Auxvaribales' to be SECOND/MONOMIAL
but when I call it using

```
[./stress_zz]
    order = SECOND
    family = MONOMIAL
    initial_from_file_var = stress_zz
   initial_from_file_timestep = 5
  [../]
```
It gives out this error message:

`Error! Trying to copy elemental solution into a variable that is not of CONSTANT MONOMIAL type.
[0] ../src/mesh/exodusII_io.C, line 671, compiled Feb 16 2021 at 16:53:25
`
I need the stress_old to calculate the stress for the next time step (incremental workflow), hence I need to import the stress from an input file.

Note that: the restart work fine with 'Variables' like disp_z
also, it works fine if 'Auxvaribales' is in CONSTANT/MONOMIAL, which I do not want.

Thanks guys!

**Comment by GiudGiud** [comment, marked as answer]
My guess would be that we can't store second order monomials in exodus files because it's not supported by this file format. 
@rwcarlsen can you please confirm/infirm?

You could try using [checkpoint](https://mooseframework.inl.gov/source/outputs/Checkpoint.html) restart instead?

**Reply by rwcarlsen** [reply]
The state of this kind of functionality has evolved gradually over the years. It should work, but you do have to set `discontinuous = true` in the exodus output section of the input file.  Also - there is this somewhat recent issue that might be relevant: https://github.com/idaholab/moose/issues/13200.

**Reply by GiudGiud** [reply]
Please let us know if `discontinuous=true` does not work, I ll add it to the issue
The restart (checkpoint) system is the better solution here, it is not limited by the limitations of exodus.

**Reply by Traiwit** [reply]
Hi guys, thank you for your suggestion, but `discontinuous = true `doesn't fix the problem.

not sure if I can use 'Checkpoint' without having to use `initial_from_file_var` and ` initial_from_file_timestep `, and just bring the whole Auxvariables from previous simulation to the new simulation

**Reply by rwcarlsen** [reply]
Did you read #13200? It seems like that might be related to your problem.
